### PR TITLE
API-2542: Hooks migration SideNav

### PR DESCRIPTION
**Pull request by johanna-adhoc**
### Story [API-2542](https://vajira.max.gov/browse/API-2542)
Migration of component SideNav from class to Hooks.

### Process
- [X ] Tests added or updated for change

### Requested feedback
All feedback is welcome

### CODE REVIEW

**Review by TS000 — DISMISSED** [review]
Looks great 🚀

File: src/components/sideNav/SideNavEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const SideNavEntry = (props: SideNavEntryProps): JSX.Element => {
+  // The isActive prop receives two arguments: a `match` object representing
+  // the original determination, and the current location. The match algorithm
+  // used by react-router only takes into account the path, and by default will
+  // include partial matches according to the https://github.com/pillarjs/path-to-regexp
+  // implementation.
+  // `navHashLinkIsActive` is both more and less strict than the default implementation:
+  // there are cases where the original would return false and this function returns true,
+  // and vice versa.
+  const navHashLinkIsActive = (pathMatch: match, location: Location): boolean => {
```
Comment (mjlumetta) [inline src/components/sideNav/SideNavEntry.tsx:27]: i think this function looks pretty substantially different now as of the React Router updates, but it's not getting picked up in the diff by virtue of moving the file.

**Review by mjlumetta — CHANGES_REQUESTED** [review]
looks good to go except there are some changes from the React Router upgrade, including some fixes to nav link highlighting, that didn't get picked up in the diff because of the side nav getting its own directory now. it's mostly in `navHashLinkIsActive` - linking the PR below. i hope it's easy to port the changes over, but let me know if you'd like to talk it through.

https://github.com/department-of-veterans-affairs/developer-portal/pull/468

**Review by TS000 — APPROVED** [review]
Nice work!

**Review by TS000 — APPROVED** [review]
Looks great! For those curious, we are creating a ticket for updating the Router to use hooks.

**Comment by johanna-adhoc** [comment]
Hey @mjlumetta !

This is the PR that we fixed the other day. Can you review it again when you have a chance? Thanks

**Review by mjlumetta — APPROVED** [review]
looks great, thanks @johanna-adhoc! 💅 🚀

**Pull request merged**